api/src/utils/sanitize-query.ts

```diff
@@ -128,7 +128,9 @@ function sanitizeFilter(rawFilter: any, accountability: Accountability | null) {
 
 function sanitizeLimit(rawLimit: any) {
 	if (rawLimit === undefined || rawLimit === null) return null;
-	return Number(rawLimit);
+	const limit = Number(rawLimit);
+	if (limit < -1) return null; // don't accept below -1
```

Comment:
I think we should also add a check for `NaN`.

```suggestion
	if (isNaN(limit) || limit < -1) return null; // don't accept NaN or below -1
```

Reply:
good suggestion, i was even thinking of adding a `Math.floor` to the result as decimals have no meaning for a `limit`.

Reply:
I would suggest adding a check using `Number.isInteger()` instead.

Reply:
Even better

Reply:
Edited! 😄

Reply:
Although slightly-unintuitively, should we intentionally not return null for NaN instead? Seems like `validateQuery` does return error when we're using `limit=a` for example:

https://github.com/directus/directus/blob/dc1a218c58110fc28ec58a43b817069ec72df173/api/src/middleware/sanitize-query.ts#L14-L24

![chrome_l6mgOJabCj](https://user-images.githubusercontent.com/42867097/209918681-4c6641bb-5b14-4812-b01c-4790f12bc7c0.png)

but returning `null` would ignore it silently. 

That being said, this does depend on what should be the expected outcome. If we want to go with "return null when NaN" for `limit`, we might have to do the same thing for `offset` and `page` to make the expected outcome of NaN values consistent across the query parameters.

Reply:
Nice catch with the tests! 💯 
Should this fix be shifted into `validateQuery()` instead? 🤔

Reply:
FWIW I do feel the lines are abit blurry, especially with this piece of logic technically also drops NaN values:

https://github.com/directus/directus/blob/dc1a218c58110fc28ec58a43b817069ec72df173/api/src/utils/sanitize-query.ts#L11-L15

but yeah let's shift to `validateQuery()` for this fix, for now. 

Looks like using `Joi.number().integer().min(-1)` to replace the following line does the integer and above -1 check we wanted, if I'm not mistaken:

https://github.com/directus/directus/blob/dc1a218c58110fc28ec58a43b817069ec72df173/api/src/utils/validate-query.ts#L14

Reply:
@br41nslug I believe https://github.com/directus/directus/pull/16722/commits/71fba5de69c6b7c8d0fe9a1dacd31db0f2a0f31e should be reverted in favor of the Joi change in `validate-query.ts` as discussed above. Apologies if it ended up being long winded and not really clear that we decided not to go with [this suggestion](https://github.com/directus/directus/pull/16722#discussion_r1058492239) 😄

Reply:
But `joi` is on the validation side right? is will only error when invalid not sanitize? Or is the idea moving away from sanitization in favor of validation 🤔Sanitization has a better user experience imo but validation is more explicit so something to be said for both approaches however it is a bit confusing that we're mixing both here

Reply:
> Sanitization has a better user experience imo but validation is more explicit so something to be said for both approaches 

I do agree that sanitization can be a better UX! Just not sure which approach should we opt for as some other query params technically aren't "sanitized" the same way (eg. `offset` and `page` aren't null-ed when we use alphabets, `export` doesn't get null-ed when it isn't one of the accepted format), at least from my current understanding.

> however it is a bit confusing that we're mixing both here

Yeah my main concern is more to do with inconsistency. If `limit` is null-ed/ignored when it is not an integer, then we may need to do the same across the query params or else they will behave slightly differently. It does feel like the existing logic (not this PR) is already somewhat blurry, so it might require a separate discussion to align the expectation here 🤔